server/s3_copy_object_action.h

```diff
@@ -38,6 +38,8 @@ enum class S3CopyObjectActionState {
   completed,                // All stages done completely
 };
 
+const u_long MaxCopyObjectSourceSize = 5368709120;  // 5GB
```

Comment:
I would recommend 
#define MaxCopyObjectSourceSize 5368709120ull
Dont insist

Reply:
I generally avoid using per-processor macros, they won't go in compiler's symbols table, and hence would be seen as a number, instead of a human readable name, when we will be debugging any core-dump in future.

Reply:
1. The standard allows compilers to treat 'const' variable of build-in type, initialized by a constant expression, as 'constexpr'. Highly likely a compiler optimizes out this variable.
2. As far as i know integer numbers are treated as int type by default.
I'm afraid that a compiler can cut the number to 1GB.
Can you add 'ul' suffix to the number.

Reply:
Checked via trivial program. gcc doesn't cut the variable.

Reply:
Sure I will change the suffix to UL.

Actually my comment was w.r.t to "#define" macro. As this is pre-proccessor macro, the macro would be replaced by its value when the code reaches compiler.